### PR TITLE
test: migrate extension-core format and lookup tests to JUnit 5

### PR DESCRIPTION
**Pull request by FrankChen021**
Part of #13948.

Builds on [PR #19875](https://github.com/apache/druid/pull/19875).

### Description

This PR migrates the JUnit 4 tests in the second extension-core batch to JUnit 5:

- avro-extensions
- parquet-extensions
- hdfs-storage
- lookups-cached-single
- kinesis-indexing-service

The tests now use Jupiter assertions, lifecycle annotations, parameterized tests, nested tests, and temporary-directory support. Transitional JUnit 4 test dependencies remain only where migrated tests inherit shared JUnit 4 fixtures from indexing-service or use the shared TestDerbyConnector utility; those dependencies can be removed after the shared fixtures are migrated.

### Validation

Ran the complete changed-test set for all five modules with Maven. The reactor completed successfully with no failures, errors, or flakes.

### Related migration PR

- Previous batch: #19875

### CODE REVIEW

File: extensions-core/hdfs-storage/src/test/java/org/apache/druid/storage/hdfs/HdfsDataSegmentKillerTest.java [hunk 1]
```diff
@@ -354,87 +353,86 @@ public String getStorageDirectory()
...
+  public void testNoStorageDirectory()
   {
-    Configuration config = new Configuration();
-    HdfsDataSegmentKiller killer = new HdfsDataSegmentKiller(
-        config,
-        new HdfsDataSegmentPusherConfig()
-        {
-          @Override
-          public String getStorageDirectory()
+    Throwable exception = assertThrows(IllegalStateException.class, () -> {
```
Comment (FrankChen021) [inline extensions-core/hdfs-storage/src/test/java/org/apache/druid/storage/hdfs/HdfsDataSegmentKillerTest.java]: [P2] Limit the exception assertion to killAll

The `assertThrows` lambda now includes filesystem setup, the successful `killer.kill(...)` path, its assertions, cleanup, and `killAll()`. If an earlier operation starts throwing the same `IllegalStateException`, this test passes early, skips the behavior it was meant to verify, and can leave `/tmp/dataSourceNew` behind. Keep setup, `kill`, assertions, and cleanup outside the lambda and wrap only `killer.killAll()`.

File: extensions-core/hdfs-storage/src/test/java/org/apache/druid/inputsource/hdfs/HdfsInputSourceTest.java [hunk 2]
```diff
@@ -95,47 +97,46 @@ public class HdfsInputSourceTest extends InitializedNullHandlingTest
...
-                     .build();
-
-      expectedException.expect(IllegalArgumentException.class);
-      expectedException.expectMessage("Only [hdfs] protocols are allowed");
-      HdfsInputSource.builder()
-                     .paths("file:/foo/bar*")
-                     .configuration(CONFIGURATION)
-                     .inputSourceConfig(DEFAULT_INPUT_SOURCE_CONFIG)
-                     .build();
+      Throwable exception = assertThrows(IllegalArgumentException.class, () -> {
```
Comment (FrankChen021) [inline extensions-core/hdfs-storage/src/test/java/org/apache/druid/inputsource/hdfs/HdfsInputSourceTest.java]: [P2] Keep the valid HDFS case outside assertThrows

The lambda first builds a valid `hdfs` input source and then the invalid `file:` source. If the valid construction regresses and throws the expected `IllegalArgumentException`, the assertion succeeds before the invalid protocol is tested. Build the allowed source before `assertThrows` and put only the rejected `file:` construction inside it.

File: extensions-core/lookups-cached-single/src/test/java/org/apache/druid/server/lookup/cache/loading/LoadingCacheTest.java [hunk 3]
```diff
@@ -50,42 +46,43 @@ public static Collection<Object[]> inputData()
...
-  @Before
-  public void setUp()
-  {
-    Assert.assertFalse(loadingCache.isClosed());
+    Assertions.assertFalse(loadingCache.isClosed());
     loadingCache.putAll(IMMUTABLE_MAP);
   }
 
-  @After
+  @AfterEach
```
Comment (FrankChen021) [inline extensions-core/lookups-cached-single/src/test/java/org/apache/druid/server/lookup/cache/loading/LoadingCacheTest.java:58]: [P3] Close each generated off-heap cache

Each parameterized method now invokes `inputData()` separately and creates a fresh `OffHeapLoadingCache`, but teardown only calls `invalidateAll()`. `close()` is what deletes the cache map from the static direct-memory MapDB, so this migration multiplies retained off-heap maps until JVM shutdown. Close the current cache in `@AfterEach`.

**Review by FrankChen021 — COMMENTED** [review]
| Severity | Findings |
| --- | ---: |
| P0 | 0 |
| P1 | 0 |
| P2 | 2 |
| P3 | 1 |
| Total | 3 |

| Severity | Findings |
| --- | ---: |
| P0 | 0 |
| P1 | 0 |
| P2 | 2 |
| P3 | 1 |
| Total | 3 |

Reviewed 54 of 54 changed files.

---

This is an automated review by Codex GPT-5.6-Sol

File: extensions-core/hdfs-storage/src/test/java/org/apache/druid/storage/hdfs/HdfsDataSegmentKillerTest.java (hunk 1 above)
Comment (FrankChen021) [reply extensions-core/hdfs-storage/src/test/java/org/apache/druid/storage/hdfs/HdfsDataSegmentKillerTest.java]: Fixed in cc8fcd7013. Filesystem setup, kill, assertions, and cleanup remain outside assertThrows; only killer.killAll() is asserted. HdfsDataSegmentKillerTest passed all 10 tests.

File: extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java [hunk 4]
```diff
@@ -241,28 +235,28 @@
...
-    Assert.assertEquals(base.getMaxRecordsPerPollConfigured(), deserialized.getMaxRecordsPerPollConfigured());
-    Assert.assertEquals(base.getMaxColumnsToMerge(), deserialized.getMaxColumnsToMerge());
+    Assertions.assertEquals(base.getAppendableIndexSpec(), deserialized.getAppendableIndexSpec());
+    Assertions.assertEquals(base.getMaxRowsInMemory(), deserialized.getMaxRowsInMemory());
+    Assertions.assertEquals(base.getMaxBytesInMemory(), deserialized.getMaxBytesInMemory());
+    Assertions.assertEquals(base.getMaxRowsPerSegment(), deserialized.getMaxRowsPerSegment());
+    Assertions.assertEquals(base.getMaxTotalRows(), deserialized.getMaxTotalRows());
+    Assertions.assertEquals(base.getIntermediatePersistPeriod(), deserialized.getIntermediatePersistPeriod());
+    Assertions.assertNull(deserialized.getBasePersistDirectory());
+    Assertions.assertEquals(base.getMaxPendingPersists(), deserialized.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java:245]: Evaluated as baseline: this deprecated getMaxPendingPersists call already existed on master before this JUnit migration. This PR only changes test framework usage, so deprecated API cleanup is separate scope and is left unchanged.

File: extensions-core/lookups-cached-single/src/test/java/org/apache/druid/server/lookup/cache/loading/LoadingCacheTest.java (hunk 3 above)
Comment (FrankChen021) [reply extensions-core/lookups-cached-single/src/test/java/org/apache/druid/server/lookup/cache/loading/LoadingCacheTest.java:58]: Fixed in cc8fcd7013. tearDown now calls close() after invalidateAll(), releasing each off-heap cache map. LoadingCacheTest passed all 18 tests.

File: extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java [hunk 5]
```diff
@@ -119,22 +113,22 @@
...
-    Assert.assertEquals(2, (int) config.getFetchThreads());
-    Assert.assertTrue(config.isSkipSequenceNumberAvailabilityCheck());
-    Assert.assertFalse(config.isResetOffsetAutomatically());
-    Assert.assertEquals(-1, config.getMaxColumnsToMerge());
+    Assertions.assertNull(config.getBasePersistDirectory());
+    Assertions.assertEquals(new OnheapIncrementalIndex.Spec(), config.getAppendableIndexSpec());
+    Assertions.assertEquals(100, config.getMaxRowsInMemory());
+    Assertions.assertEquals(100, config.getMaxRowsPerSegment().intValue());
+    Assertions.assertEquals(new Period("PT1H"), config.getIntermediatePersistPeriod());
+    Assertions.assertEquals(100, config.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java:121]: Evaluated as baseline: this deprecated getMaxPendingPersists call already existed on master before this JUnit migration. This PR only changes test framework usage, so deprecated API cleanup is separate scope and is left unchanged.

File: extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java [hunk 6]
```diff
@@ -68,24 +62,24 @@
...
-    Assert.assertEquals(5000, config.getRecordBufferFullWait());
-    Assert.assertNull(config.getFetchThreads());
-    Assert.assertFalse(config.isSkipSequenceNumberAvailabilityCheck());
-    Assert.assertFalse(config.isResetOffsetAutomatically());
+    Assertions.assertNull(config.getBasePersistDirectory());
+    Assertions.assertEquals(new OnheapIncrementalIndex.Spec(), config.getAppendableIndexSpec());
+    Assertions.assertEquals(150000, config.getMaxRowsInMemory());
+    Assertions.assertEquals(5_000_000, config.getMaxRowsPerSegment().intValue());
+    Assertions.assertEquals(new Period("PT10M"), config.getIntermediatePersistPeriod());
+    Assertions.assertEquals(0, config.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java:70]: Evaluated as baseline: this deprecated getMaxPendingPersists call already existed on master before this JUnit migration. This PR only changes test framework usage, so deprecated API cleanup is separate scope and is left unchanged.

File: extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java:245]: Evaluated as baseline: this deprecated getMaxPendingPersists call already existed on master before this JUnit migration. This PR only changes test framework usage, so deprecated API cleanup is separate scope and is left unchanged.

File: extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java [hunk 7]
```diff
@@ -176,31 +170,31 @@
...
-    Assert.assertEquals(base.getMaxColumnsToMerge(), deserialized.getMaxColumnsToMerge());
+    Assertions.assertEquals(null, deserialized.getExtra());
+    Assertions.assertEquals(base.getAppendableIndexSpec(), deserialized.getAppendableIndexSpec());
+    Assertions.assertEquals(base.getMaxRowsInMemory(), deserialized.getMaxRowsInMemory());
+    Assertions.assertEquals(base.getMaxBytesInMemory(), deserialized.getMaxBytesInMemory());
+    Assertions.assertEquals(base.getMaxRowsPerSegment(), deserialized.getMaxRowsPerSegment());
+    Assertions.assertEquals(base.getMaxTotalRows(), deserialized.getMaxTotalRows());
+    Assertions.assertEquals(base.getIntermediatePersistPeriod(), deserialized.getIntermediatePersistPeriod());
+    Assertions.assertNull(deserialized.getBasePersistDirectory());
+    Assertions.assertEquals(base.getMaxPendingPersists(), deserialized.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java:181]: Evaluated as baseline: this deprecated getMaxPendingPersists call already existed on master before this JUnit migration. This PR only changes test framework usage, so deprecated API cleanup is separate scope and is left unchanged.

File: extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisorTuningConfigTest.java [hunk 8]
```diff
@@ -55,20 +55,20 @@
...
-    Assert.assertEquals(8L, (long) config.getChatRetries());
-    Assert.assertEquals(Duration.standardSeconds(10), config.getHttpTimeout());
-    Assert.assertEquals(Duration.standardSeconds(80), config.getShutdownTimeout());
-    Assert.assertEquals(Duration.standardSeconds(120), config.getRepartitionTransitionDuration());
+    Assertions.assertNull(config.getBasePersistDirectory());
+    Assertions.assertEquals(new OnheapIncrementalIndex.Spec(), config.getAppendableIndexSpec());
+    Assertions.assertEquals(150000, config.getMaxRowsInMemory());
+    Assertions.assertEquals(5_000_000, config.getMaxRowsPerSegment().intValue());
+    Assertions.assertEquals(new Period("PT10M"), config.getIntermediatePersistPeriod());
+    Assertions.assertEquals(0, config.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisorTuningConfigTest.java:63]: Evaluated as baseline: this deprecated getMaxPendingPersists call already existed on master before this JUnit migration. This PR only changes test framework usage, so deprecated API cleanup is separate scope and is left unchanged.

File: extensions-core/hdfs-storage/src/test/java/org/apache/druid/inputsource/hdfs/HdfsInputSourceTest.java (hunk 2 above)
Comment (FrankChen021) [reply extensions-core/hdfs-storage/src/test/java/org/apache/druid/inputsource/hdfs/HdfsInputSourceTest.java]: Fixed in cc8fcd7013. The valid hdfs source is built before assertThrows, and the lambda contains only the rejected file: source construction. HdfsInputSourceTest passed all 35 nested tests.

File: extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisorTuningConfigTest.java [hunk 9]
```diff
@@ -101,18 +101,18 @@
...
-    Assert.assertEquals(14L, (long) config.getChatRetries());
-    Assert.assertEquals(Duration.standardSeconds(15), config.getHttpTimeout());
-    Assert.assertEquals(Duration.standardSeconds(95), config.getShutdownTimeout());
-    Assert.assertEquals(Duration.standardSeconds(500), config.getRepartitionTransitionDuration());
+    Assertions.assertNull(config.getBasePersistDirectory());
+    Assertions.assertEquals(new OnheapIncrementalIndex.Spec(), config.getAppendableIndexSpec());
+    Assertions.assertEquals(100, config.getMaxRowsInMemory());
+    Assertions.assertEquals(100, config.getMaxRowsPerSegment().intValue());
+    Assertions.assertEquals(new Period("PT1H"), config.getIntermediatePersistPeriod());
+    Assertions.assertEquals(100, config.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisorTuningConfigTest.java:109]: Evaluated as baseline: this deprecated getMaxPendingPersists call already existed on master before this JUnit migration. This PR only changes test framework usage, so deprecated API cleanup is separate scope and is left unchanged.

File: extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java [hunk 10]
```diff
@@ -333,26 +332,26 @@
...
-    Assert.assertEquals(new Period().withDays(Integer.MAX_VALUE), copy.getIntermediateHandoffPeriod());
-    Assert.assertEquals(-1, copy.getMaxColumnsToMerge());
+    Assertions.assertEquals(original.getAppendableIndexSpec(), copy.getAppendableIndexSpec());
+    Assertions.assertEquals(1, copy.getMaxRowsInMemory());
+    Assertions.assertEquals(3, copy.getMaxBytesInMemory());
+    Assertions.assertEquals(2, copy.getMaxRowsPerSegment().intValue());
+    Assertions.assertEquals(100L, (long) copy.getMaxTotalRows());
+    Assertions.assertEquals(new Period("PT3S"), copy.getIntermediatePersistPeriod());
+    Assertions.assertNull(copy.getBasePersistDirectory());
+    Assertions.assertEquals(4, copy.getMaxPendingPersists());
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java:342]: Evaluated as baseline: this deprecated getMaxPendingPersists call already existed on master before this JUnit migration. This PR only changes test framework usage, so deprecated API cleanup is separate scope and is left unchanged.

File: extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskTuningConfigTest.java:181]: Evaluated as baseline: this deprecated getMaxPendingPersists call already existed on master before this JUnit migration. This PR only changes test framework usage, so deprecated API cleanup is separate scope and is left unchanged.

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the updated code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 54 of 54 changed files. All three prior findings are resolved: both exception assertions are scoped to the rejected operation, and each generated off-heap cache is closed during teardown.

---

This is an automated review by Codex GPT-5.6-Sol

File: extensions-core/lookups-cached-single/src/test/java/org/apache/druid/server/lookup/PollingLookupSerDeserTest.java [hunk 11]
```diff
@@ -41,40 +40,40 @@
...
   }
 
   @JsonTypeName("mock")
   private static class MockDataFetcher implements DataFetcher
   {
     @JsonCreator
-    public MockDataFetcher()
+    public void initPollingLookupSerDeserTest()
     {
     }
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/lookups-cached-single/src/test/java/org/apache/druid/server/lookup/PollingLookupSerDeserTest.java:78]: Actionable. The JUnit migration had converted the original no-argument MockDataFetcher constructor into an instance void method while retaining @JsonCreator. I restored it as public MockDataFetcher(), so @JsonCreator now targets a valid constructor and the intended Jackson subtype behavior is preserved. PollingLookupSerDeserTest passed both parameterized cases (2 tests, 0 failures); Checkstyle and SpotBugs also passed.

File: extensions-core/parquet-extensions/pom.xml [hunk 12]
```diff
@@ -127,10 +127,25 @@
       <groupId>com.github.rvesse</groupId>
       <artifactId>airline</artifactId>
       <scope>provided</scope>
+    </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter</artifactId>
+      <scope>test</scope>
+    </dependency>
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/parquet-extensions/pom.xml]: Fixed in commit 4676b86270 by normalizing the newly added JUnit dependency block to the surrounding two-space POM indentation. Checkstyle passed with 0 violations, and the module-scoped SpotBugs check passed with 0 bugs/errors.

File: extensions-core/hdfs-storage/pom.xml [hunk 13]
```diff
@@ -106,9 +106,24 @@
         </dependency>
 
         <!-- Tests -->
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <scope>test</scope>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply extensions-core/hdfs-storage/pom.xml]: Fixed in commit 4676b86270 by normalizing the newly added JUnit dependency block to the surrounding four-space POM indentation. Checkstyle passed with 0 violations, and the module-scoped SpotBugs check passed with 0 bugs/errors.

**Review by GWphua — APPROVED** [review]
